pkg/gossip/server.go

```diff
@@ -370,11 +373,10 @@ func (s *server) start(addr net.Addr) {
 	broadcast := func() {
 		// Close the old ready and open a new one. This will broadcast to all
 		// receivers and setup a fresh channel to replace the closed one.
-		s.mu.Lock()
-		defer s.mu.Unlock()
-		ready := make(chan struct{})
-		close(s.mu.ready)
-		s.mu.ready = ready
+		ready := s.ready.Load().(chan struct{})
```

Comment:
I was wrong about the order of closing and storing. We actually want the opposite order, so that the load of the channel after the previous one is signaled is guaranteed to read the new channel. So something like:
```go
close(s.ready.Swap(make(chan struct{})).(chan struct{}))
```